react/src/components/Chat/ChatMessageContent.tsx

```diff
@@ -1,89 +1,163 @@
+import Flex from '../Flex';
 import CopyButton from './CopyButton';
-import { Card, Typography } from 'antd';
+import { SyntaxHighlighter, SyntaxHighlighterProps } from './SyntaxHighight';
+import { theme, Typography } from 'antd';
 // `rehype-katex` does not import the CSS file, so we need to import it manually.
 import 'katex/dist/katex.min.css';
-import React from 'react';
+import { marked } from 'marked';
+import React, { memo, useCallback, useMemo } from 'react';
 import Markdown from 'react-markdown';
-import {
-  Prism as SyntaxHighlighter,
-  SyntaxHighlighterProps,
-} from 'react-syntax-highlighter';
 import { oneLight } from 'react-syntax-highlighter/dist/esm/styles/prism';
 import rehypeKatex from 'rehype-katex';
 import remarkGfm from 'remark-gfm';
 import remarkMath from 'remark-math';
 
 const { Text } = Typography;
 
-const ChatMessageContent: React.FC<{
-  children?: string;
-}> = ({ children }) => {
-  return (
-    <Markdown
-      remarkPlugins={[remarkGfm, remarkMath]}
-      rehypePlugins={[rehypeKatex]}
-      components={{
-        p({ node, ...props }) {
-          // @ts-ignore
-          return <p {...props} style={{ whiteSpace: 'pre-wrap' }} />;
-        },
-        code(props) {
-          const { children, className, node, ref, ...rest } = props;
-          const match = /language-(\w+)/.exec(className || '');
-          const content = String(children).replace(/\n$/, '');
-          return match ? (
-            <Card
-              title={
-                <Text style={{ fontWeight: 'normal' }} type="secondary">
-                  {match[1]}
-                </Text>
-              }
-              type="inner"
-              size="small"
+function parseMarkdownIntoBlocks(markdown: string): string[] {
+  const tokens = marked.lexer(markdown);
+  return tokens.map((token) => token.raw);
+}
+
+const CodeBlock = memo<SyntaxHighlighterProps>(({ children, ...props }) => (
+  <SyntaxHighlighter {...props}>{children}</SyntaxHighlighter>
+));
+
+CodeBlock.displayName = 'CodeBlock';
+
+const CodeHead = memo<{ lang: string; extra?: React.ReactNode }>(
+  ({ lang, extra }) => {
+    const { token } = theme.useToken();
+
+    return (
+      <Flex
+        style={{
+          margin: '0',
+          minHeight: '38px',
```

Comment:
You can use `number` directly in style object.
```suggestion
          margin: 0,
          minHeight: 38,
```